exercises/practice/robot-name/uRobotName.pas

```diff
@@ -0,0 +1,40 @@
+unit uRobotName;
+
+interface
+
+uses
+  Generics.Collections;
+
+type
+  TRobot = class
+  private
+    class var NamesTracking: TList<string>; // global var to track names already used
+    var fName: string;
```

Comment:
I am not a fan of this.  This, to me, is steering the learner towards a particular solution.  This exercise is a more difficult one so I am not keen on pointing them in any particular direction.  There are only a finite number of names available, keeping track of actual names is not necessary to solve this exercise.